src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -334,10 +409,7 @@ private void AddBindings(ArrayBuilder<BoundStatement> sectionBuilder, ImmutableA
                     {
                         var source = kv.Key;
                         var dest = kv.Value;
-                        var rewriter = this.LocalRewriter;
-                        sectionBuilder.Add(_factory.ExpressionStatement(
-                            rewriter.MakeStaticAssignmentOperator(
-                                _factory.Syntax, rewriter.VisitExpression(dest), rewriter.VisitExpression(source), RefKind.None, dest.Type, false)));
+                        sectionBuilder.Add(_factory.Assignment(dest, source));
```

Comment:
`DecisionTreeBuilder` is the base class. With the exception of constant values from patterns (for which only the constant value is used, not the bound node), it does not have access to expressions from the initial bound tree.